structs/src/strg.rs

```diff
@@ -37,6 +37,68 @@ pub struct Strg<'r>
 
 impl<'r> Strg<'r>
 {
+    fn is_jap_version(languages: Vec<FourCC>) -> bool
+    {
+        languages.len() == 2 && languages.iter().any(|lang| lang.as_bytes() == b"ENGL") && languages.iter().any(|lang| lang.as_bytes() == b"JAPN")
+    }
+
+    pub fn add_strings_2(self: &mut Self, strings: Vec<String>, languages: Vec<FourCC>)
+    {
+        let is_jap = Self::is_jap_version(languages.clone());
+        for table in self.string_tables.as_mut_vec().iter_mut() {
+            if languages.contains(&table.lang) {
+                for string in strings.iter() {
+                    if is_jap {
+                        table.strings.as_mut_vec().push(format!("&line-extra-space=4;&font=C29C51F1;{}", string).into());
+                    } else {
+                        table.strings.as_mut_vec().push(format!("{}", string).into());
+                    }
+                }
+            }
+        }
+    }
+
+    pub fn add_strings(self: &mut Self, strings: Vec<String>)
+    {
+        let supported_languages = vec![
+            FourCC::from(b"ENGL"),
+            FourCC::from(b"DUTC"),
+            FourCC::from(b"FREN"),
+            FourCC::from(b"GERM"),
+            FourCC::from(b"ITAL"),
+            FourCC::from(b"JAPN"),
+            FourCC::from(b"SPAN"),
+        ];
+        self.add_strings_2(strings, supported_languages);
+    }
+
+    pub fn edit_strings_2(self: &mut Self, (from, to): (String, String), languages: Vec<FourCC>)
```

Comment:
Same as above. You should also use a shared constant vec for the supported languages. If the compiler complains, you can use [lazy_static](https://docs.rs/lazy_static/latest/lazy_static/)